presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -735,9 +740,23 @@ public synchronized Map<String, Optional<Partition>> getPartitionsByNames(Metast
         }
         Map<String, Optional<Partition>> delegateResult = delegate.getPartitionsByNames(metastoreContext, databaseName, tableName, partitionNamesToQuery.build());
         resultBuilder.putAll(delegateResult);
+
+        cacheLastDataCommitTimes(delegateResult, databaseName, tableName);
+
         return resultBuilder.build();
     }
 
+    public void cacheLastDataCommitTimes(Map<String, Optional<Partition>> existingPartitions, String databaseName, String tableName)
```

Comment:
private static void storeLastDataCommitTimes

Reply:
Can be private; but this function cannot be static since it will access the variable lastDataCommitTimesForRead of the instance.